### ISSUE TITLE
Iterable SDK hanging my app

### ISSUE THREAD

**Issue by jbelkins** [Iterable/swift-sdk#513]
About 1/2 of my production crashes right now are from my app hanging on startup, on this line of code which is called when starting the Iterable SDK:
https://github.com/Iterable/swift-sdk/blob/master/swift-sdk/Internal/IterableAppIntegrationInternal.swift#L29
(I have to add, your spelling of `semaphore` is not inspiring a lot of confidence, lol.)

This affects less than 0.5% of my sessions, but it is still high enough to get my attention.  Crashes occur on iPhone & iPad devices running various versions of iOS 14 & 15.  My app was running Iterable iOS SDK v6.3.3 for all of these crashes.

Here is a full stack trace from my crash reporter:
```
App Hang: The app was terminated while unresponsive

0  libsystem_kernel.dylib _semaphore_wait_trap
1  libdispatch.dylib      __dispatch_sema4_wait
2  libdispatch.dylib      __dispatch_semaphore_wait_slow
3  Trala                  notificationsEnabled.get (IterableAppIntegrationInternal.swift:29:24)
4  Trala                  notificationsEnabled.get (<compiler-generated>)
5  Trala                  register (InternalIterableAPI.swift:152:31)
6  Trala                  register (IterableAPI.swift:208:33)
7  Trala                  application (AppDelegate.swift:126:21)
8  Trala                  application (<compiler-generated>)
9  libdispatch.dylib      __dispatch_call_block_and_release
10 libdispatch.dylib      __dispatch_client_callout
11 libdispatch.dylib      __dispatch_main_queue_callback_4CF
12 CoreFoundation         ___CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
13 CoreFoundation         ___CFRunLoopRun
14 CoreFoundation         _CFRunLoopRunSpecific
15 GraphicsServices       _GSEventRunModal
16 UIKitCore              -[UIApplication _run]
17 UIKitCore              _UIApplicationMain
18 Trala                  main (main.swift:29:1)
19 <unknown>              0x105315a20
```

Please let me know if you need more info.

**Comment by jbelkins** [comment]
Apple's documentation for UNUSerNotificationCenter's `getNotificationSettings(completion:)` method says the completion block "_may_ execute [the completion] block on a background thread" (my emphasis on "may".)

I would read this to imply that the completion block may be called on the main thread under some circumstances, which would explain the deadlock that apparently is happening in my app.

In any case, it's poor practice to wait the main thread on a semaphore while stuff happens on other threads, and this code should be changed to not block the main thread that way.

**Comment by tapashmajumder** [comment]
@jbelkins Thank you for brining this to our attention. We have not seen this issue locally. 

We will greatly appreciate if you can try our solution and provide feedback once we commit this to a branch.

**Comment by tapashmajumder** [comment]
@jbelkins Agreed. Our code earlier was non-blocking and this code was introduced via due to some other problems. We will be looking at this issue with high priority.

**Comment by jbelkins** [comment]
Thanks for your prompt attention.  Please tag me when you publish a fix and I'll ship it as soon as I can.
FYI I just filed an Iterable support ticket citing this Github issue as well.

**Comment by tapashmajumder** [comment]
Thanks. The ticket is implemented and is being reviewed. We will be publishing a fix soon.

**Comment by tapashmajumder** [comment]
@jbelkins We have the fix merged in `master` branch. Would you be able to try that?

**Comment by jbelkins** [comment]
@tapashmajumder Might be middle of next week before I can put up a new app release.  If you can get this tested at your end, you should go ahead & release it.

And if I do ship this fix before it's published, rather than take everything in master, I'll lock to this commit: https://github.com/Iterable/swift-sdk/commit/bccfb50c8a9be06446de340f139a439420f0aa80 Would that work?

**Comment by tapashmajumder** [comment]
@jbelkins Yes that is the right commit if you can release. 

Master currently has a security fix which we would like to release together with other security tickets that we are currently working on. So it may take us some time (probably around 7-10 days or so) to publish our next release.

**Comment by jbelkins** [comment]
@tapashmajumder Just shipped this fix today.  I should be able to say whether this is fixed in about a week or so.

**Comment by tapashmajumder** [comment]
Hi @jbelkins Thank you so much. Please keep us updated. We will be releasing a new version with this and a few other fixes in a week or two.

**Comment by jbelkins** [comment]
I’ve had this fix in production for about 9 days now and no reoccurrences thus far.  I think this can be closed now.

Please let me know when this ships in a stable release.

**Comment by tapashmajumder** [comment]
@jbelkins That is great news. I will update when we ship this change.

**Comment by tapashmajumder** [comment]
This issue is fixed in latest release 6.4.0. Closing.
